plugins/modules/zos_zfs_resize.py

```diff
@@ -0,0 +1,427 @@
+        size=dict(type="int", required=True),
+        size_type=dict(
+                type="str",
+                required=False,
+                choices=["k", "m", "g", "cyl", "trk"],
+                default="k",
+            ),
+        no_auto_increment=dict(type="bool", required=False, default=False),
+        verbose=dict(type="bool", required=False, default=False),
+    )
+
+    try:
+        parser = better_arg_parser.BetterArgParser(args_def)
+        parsed_args = parser.parse_args(module.params)
+        module.params = parsed_args
+    except ValueError as err:
+        module.fail_json(
+            msg='Parameter verification failed.',
+            stderr=str(err)
+        )
+
+    res_args = dict()
+    target = module.params.get("target")
+    size = module.params.get("size")
+    size_type = module.params.get("size_type")
+    noai = module.params.get("no_auto_increment")
+    verbose = module.params.get("verbose")
+    changed = False
+
+    #Validation to found target on the system and also get the mount_point
+    mount_target = found_mount_target(module=module, target=target)
+
+    #Initialize the class with the target
+    aggregate_name = zfsadm(aggregate_name=target, module=module)
+
+    rc, stdout, stderr = aggregate_name.get_agg_size()
+
+    if rc == 0:
+        old_size, old_free = get_size_and_free(string=stdout)
+
+    res_args.update(
+        dict(
+            target=target,
+            mount_target=mount_target,
+            old_size=old_size,
+            old_free=old_free,
+            size=size,
+            cmd="",
+            changed=changed,
+            rc=0,
+            stdout="",
+            stderr="",
+        )
+    )
+
+    if size_type != "k":
+        size = calculate_size_on_k(size=size, size_type=size_type)
+
+    #Validations to know witch function will be execute
+    operation = ""
+    minimum_size_t_shrink = old_size - old_free
+
+    if size == old_size:
+        module.fail_json(msg="Same size declared with the size of the zfz")
+    elif size > old_size:
+        operation = "grow"
+    elif size >= minimum_size_t_shrink and size < old_size:
+        operation = "shrink"
+    else:
+        module.fail_json(msg="Error code 119 not enough space to shrink")
+
+    path = module._remote_tmp if verbose else ""
+
+    cmd, tmp_file = create_command(size=size, noai=noai, verbose=path)
+
+    #Execute the function
+    rc, stdout, stderr, cmd = aggregate_name.grow_shrink(operation=operation, cmd=cmd)
+
+    if rc == 0:
+        changed = True
+
+        rc_size, stdout_size, stderr_size = aggregate_name.get_agg_size()
+        if rc_size == 0:
+            new_size, new_free = get_size_and_free(string=stdout_size)
+
+        if verbose:
+            output = get_full_output(file=tmp_file, module=module)
+            res_args.update(
+                verbose_output=output,
+            )
+    else:
+        if verbose:
+            os.remove(tmp_file)
```

Comment:
I know tmp_file is being removed inside get_full_output, I would recommend deleting the tmp_file in this flow so other people reading the code will know for sure, instead of checking that other function.